modules/core/kivent_core/rendering/animation.pyx

```diff
@@ -0,0 +1,87 @@
+from kivent_core.managers.resource_managers cimport ModelManager
+from kivent_core.managers.resource_managers import texture_manager
+from kivent_core.rendering.model cimport VertexModel
+from kivent_core.memory_handlers.membuffer cimport Buffer
+from kivent_core.memory_handlers.block cimport MemoryBlock
+
+
+cdef class Frame:
+
+    def __cinit__(self, ModelManager model_manager):
+        self.model_manager = model_manager
+
+    property model:
+        def __get__(self):
+            cdef VertexModel model = <VertexModel>self.frame_pointer.model
+            return model.name
+
+        def __set__(self, value):
+            self.frame_pointer.model = <void*>self.model_manager.models[value]
+
+    property texture:
+        def __get__(self):
+            return texture_manager.get_texname_from_texkey(self.frame_pointer.texkey)
+
+        def __set__(self, str value):
+            self.frame_pointer.texkey = texture_manager.get_texkey_from_texname(value)
+
+    property duration:
+        def __get__(self):
+            return self.frame_pointer.duration
+
+        def __set__(self, unsigned int value):
+            self.frame_pointer.duration = value
+
+
+cdef class FrameList:
+
+    def  __cinit__(self, frame_count, frame_buffer, model_manager, name):
+        self._frame_count = frame_count
+        self.model_manager = model_manager
+        self.name = name
+
+        cdef MemoryBlock frames_block = MemoryBlock(
+            frame_count*sizeof(FrameStruct), sizeof(FrameStruct), 1)
+        frames_block.allocate_memory_with_buffer(frame_buffer)
```

Comment:
Do you have the raw traceback? (you can debug with gdb --args python XXX, then "r" to start it, and "bt" when it crash)